src/js/states/mods.js

```diff
@@ -132,6 +145,32 @@ export class ModsState extends TextualGameState {
         this.app.platformWrapper.openExternalLink(THIRDPARTY_URLS.modBrowser);
     }
 
+    toggleMod(checkbox, modID, toggleIsOn) {
+        if (toggleIsOn()) {
+            MODS.disableMod(modID);
+        } else {
+            MODS.enableMod(modID);
+        }
+        checkbox.classList.toggle("checked");
+    }
+
+    applyModToggleDialog() {
+        const signals = this.dialogs.showWarning(
+            T.dialogs.applyModsWarning.title,
+            T.dialogs.applyModsWarning.desc,
+            ["cancel:good", "continue:bad"]
+        );
+        return new /** @type {typeof Promise<void>} */ (Promise)(resolve => {
+            signals.continue.add(resolve);
+        });
+    }
+
+    async applyModToggle() {
+        this.applyModToggleDialog().then(_ => {
+            MODS.applyModEnableDisable();
+        });
+    }
```

Comment:
this seems extremely un-idiomatic (why not just combine the functions and add the callback directly to `signals.continue`?)

Reply:
If I'm being honest I don't really know how to do that. I just copied the code for the coming soon dialog.